lib/puppet/pops/evaluator/runtime3_support.rb

```diff
@@ -267,11 +267,11 @@ def external_call_function(name, args, scope, &block)
 
   def call_function(name, args, o, scope, &block)
     file, line = extract_file_line(o)
-    loader = Adapters::LoaderAdapter.loader_for_model_object(o, scope)
+    loader = Adapters::LoaderAdapter.loader_for_model_object(o, scope, file)
     if loader && func = loader.load(:function, name)
       Puppet::Util::Profiler.profile(name, [:functions, name]) do
         # Add stack frame when calling
-        return Puppet::Pops::PuppetStack.stack(file, line, func, :call, [scope, *args], &block)
+        return Kernel.eval('func.call(scope, *args, &block)', Kernel.binding, file || '', line)
```

Comment:
hm, this is not ideal as it is much harder to understand when taken out of the context of Puppet::Stack. Understand why this moved here. Suggest a comment stating `# See Puppet::Pops::PuppetStack` as the relationship is otherwise lost to the reader.